internal/services/applications/parse/api_access.go

```diff
@@ -76,3 +76,17 @@ func (id ApiAccessId) Segments() []resourceids.Segment {
 func (id ApiAccessId) String() string {
 	return fmt.Sprintf("Application API Access (Application ID: %q, API Client ID: %q)", id.ApplicationId, id.ApiClientId)
 }
+
+func (id ApiAccessId) FromParseResult(input resourceids.ParseResult) error {
```

Comment:
This'll need to be a pointer:

```suggestion
func (id *ApiAccessId) FromParseResult(input resourceids.ParseResult) error {
```